### ISSUE TITLE
[Proposal] Use Elements Wallet as Vault 

### ISSUE THREAD

**Issue by tiero** [TDex-network/tdex-daemon#294]
This is a competing idea with https://github.com/TDex-network/tdex-daemon/issues/293

### TL;DR 

Use Elements directly as vault for markets and fee account 

### Long (?) Story

In line of "tdex-daemon should focus entirely on Atomic Swap & Trading, not on solving blockchain wallet indexing" expresses in https://github.com/TDex-network/tdex-daemon/issues/260 this moves the concept forward: Only care about swap transaction construction, pricing startegies and utxo fragmentation (for concurren requests)
This will need a profound refactor of the system, basically droppoing entirely unspents and vault repositories, adding Elements as depenedency and invoke on demand RPCs such as `createrawtransaction`, `importmulti` && `importblindingkey` (for swap proposer), `blindrawtransaction`, `signrawtransactionwithwallet`

Main benefit is we inherith all future Elements protcol changes such as PSET2 spec. 

### TODOs

Validate such complex swap transaction could be done with Elements node entirely 

### Problems

- We depend on Elements and inherith his bootlenecks (if any)
- Hard to backup/keep in sync wallet.dat file
- We need a PSBT1 -> raw transaction converter for the time being

**Comment by tiero** [comment]
I do think is nice to have a competing implementation having this elements oriented approach. working here :arrow_right:  https://github.com/tiero/zion
